easybuild/framework/easyblock.py

```diff
@@ -2156,15 +2171,15 @@ def _sanity_check_step(self, custom_paths=None, custom_commands=None, extension=
                 elif not isinstance(xs, tuple):
                     raise EasyBuildError("Unsupported type '%s' encountered in %s, not a string or tuple",
                                          key, type(xs))
-                found = False
-                for name in xs:
-                    path = os.path.join(self.installdir, name)
-                    if check_fn(path):
-                        self.log.debug("Sanity check: found %s %s in %s" % (typ, name, self.installdir))
-                        found = True
-                        break
-                    else:
-                        self.log.debug("Could not find %s %s in %s" % (typ, name, self.installdir))
+
+                found = check_path(xs, typ, check_fn)
+
+                # for library files in lib/, also consider fallback to lib64/ equivalent
+                if not found and build_option('lib64_fallback_sanity_check'):
+                    if all(x.startswith('lib/') for x in xs):
```

Comment:
I don't understand this `all`? why do all paths have to start with `lib/`?

Reply:
a wait, xs is a tuple which we do OR?

Reply:
Yes, `xs` is one entry from e.g. the `files` list from `sanity_check_paths` can be a tuple (and if it's just a string, we make it a tuple with a single element higher up).